src/solver/LexLSLeastSquareSolver.cpp

```diff
@@ -0,0 +1,274 @@
+  big_number_(options.big_number().value())
+{
+  LexLS::ParametersLexLSI param;
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(max_number_of_factorizations, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(max_number_of_factorizations, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(tol_linear_dependence, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(tol_wrong_sign_lambda, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(tol_correct_sign_lambda, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(tol_feasibility, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(regularization_type, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(max_number_of_CG_iterations, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(variable_regularization_factor, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(cycling_handling_enabled, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(cycling_max_counter, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(cycling_relax_step, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(output_file_name, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(modify_x_guess_enabled, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(modify_type_active_enabled, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(modify_type_inactive_enabled, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(set_min_init_ctr_violation, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(use_phase1_v0, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(log_working_set_enabled, param);
+  TVM_PROCESS_OPTION_PUBLIC_ACCESS(deactivate_first_wrong_sign, param);
+
+  solver_.setParameters(param);
+}
+
+void LexLSLeastSquareSolver::initializeBuild_(int nObj, int nEq, int nIneq, bool)
+{
+  resize_(nObj, nEq, nIneq, true);
+
+  autoMinNorm_ = false;
+}
+
+LexLSLeastSquareSolver::ImpactFromChanges LexLSLeastSquareSolver::resize_(int nObj, int nEq, int nIneq, bool)
+{
+  int n = variables().totalSize();
+  int nCstr = nEq + nIneq;
+  ImpactFromChanges impact;
+
+  impact.objectives_ = ImpactFromChanges::willReallocate(data2_, nObj, n + 2);
+  data2_.resize(nObj, n + 2);
+  data2_.setZero();
+  new(&A2_) MatrixXdBlock(data2_.leftCols(n));
+  new(&l2_) MatrixXdCol(data2_.col(n));
+  new(&u2_) MatrixXdCol(data2_.col(n + 1));
+  if(autoMinNorm_)
+  {
+    A2_.setIdentity();
+    l2_.setZero();
+    u2_.setZero();
+  }
+  else
+  {
+    A2_.setZero();
+    l2_.setConstant(-big_number_);
+    u2_.setConstant(+big_number_);
+  }
+  impact.equalityConstraints_ = ImpactFromChanges::willReallocate(data1_, nCstr, n + 2);
+  data1_.resize(nCstr, n + 2);
+  new(&A1_) MatrixXdBlock(data1_.leftCols(n));
+  new(&l1_) MatrixXdCol(data1_.col(n));
+  new(&u1_) MatrixXdCol(data1_.col(n + 1));
+  A1_.setZero();
+  l1_.setConstant(-big_number_);
+  u1_.setConstant(+big_number_);
+  impact.bounds_ = ImpactFromChanges::willReallocate(data0_, n, 2);
+  data0_.resize(n, 2);
+  new(&xl_) MatrixXdCol(data0_.col(0));
+  new(&xu_) MatrixXdCol(data0_.col(1));
+  xl_.setConstant(-big_number_);
+  xu_.setConstant(+big_number_);
+  x0_ = Eigen::VectorXd::Zero(n);
```

Comment:
We should be in this method when something on the size of the problem has changed (either the size of the variable or the size of constraints).
It is difficult to follow how this affect the active set and the solution, so it's safer to reset (both the variable value and the active set).
If we really see a negative impact on performance (it could be positive when constraints are moved/rearranged/shuffle), we can look at improving this later.